### PR TITLE
feat: Add `useLanguageDirection` hook

### PR DESCRIPTION
**Pull request by A7med3bdulBaset**
This PR adds a new hook `useLanguageDirection` that returns the direction of the current locale; `ltr` or `rtl`. There are too many use cases for this hook, like setting html dir attribute, styling components based on the direction, etc. 

This PR also adds documentation, examples, and tests for the new hook.

The `useLanguageDirection` name is NOT final, it can be changed if there is a better name. such as `useDirection`, `useLangDirection`, `useLocaleDirection`, etc.

### CODE REVIEW

File: examples/example-next-13-next-auth/src/app/layout.tsx [hunk 1]
```diff
@@ -1,12 +1,17 @@
...
   children: ReactNode;
 };
 
-// Even though this component is just passing its children through, the presence
-// of this file fixes an issue in Next.js 13.3.0 where link clicks that switch
-// the locale would otherwise be ignored.
-export default function RootLayout({children}: Props) {
-  return children;
+export default function Layout({children}: Props) {
+  const locale = useLocale();
```
Comment (amannn) [inline examples/example-next-13-next-auth/src/app/layout.tsx:9]: As a side note, `useLocale` will likely not work outside of the `[locale]` folder when static rendering support lands in next-intl based on `createServerContext`.

File: packages/use-intl/src/react/useLanguageDirection.tsx [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import useIntlContext from './useIntlContext';
+
+export default function useLanguageDirection(): 'ltr' | 'rtl' {
```
[the remark this answers is not in the document]
Comment (A7med3bdulBaset) [reply packages/use-intl/src/react/useLanguageDirection.tsx:3]: I agree with you. `useTextDirection` is the most indicative.

File: examples/example-next-13-next-auth/src/app/layout.tsx (hunk 1 above)
Comment (A7med3bdulBaset) [reply examples/example-next-13-next-auth/src/app/layout.tsx:9]: If `useLocale` will not work outside the context, `useTextDirection` depends on the same approach so it won't work either.

File: docs/pages/docs/usage/configuration.mdx [hunk 3]
```diff
@@ -142,3 +142,44 @@ function Component() {
   const timeZone = useTimeZone();
 }
 ```
+
+{/* useLanguageDirection */}
+## Language direction
```
[the remark this answers is not in the document]
Comment (A7med3bdulBaset) [reply docs/pages/docs/usage/configuration.mdx:147]: Your arguments make more sense than mine. I had to study all these side notes before pushing a PR. I propose to close this PR. Then you provide the right way to implement it: a documentation note or a built-in method.

**Comment by amannn** [comment]
Thank you for your feedback @A7med3bdulBaset! Based on your PR, I've set up https://github.com/amannn/next-intl/pull/380 to document RTL support.  

Would you be interested in reviewing the PR and providing feedback?

As a side note: I experimented with bringing proper ESM support to next-intl this week, so we can have proper tree shaking, but unfortunately I think we have to wait for Next.js to provide ESM support for the APIs they currently expose. So that might be another argument for suggesting a user-land implementation, at least for the time being.

Thanks again! 🙌

**Pull request closed without merging**